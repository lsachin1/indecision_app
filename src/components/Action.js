import React from 'react';

export class Action extends React.Component {

	constructor(props){
		super(props);
		this.handlePick = this.handlePick.bind(this);
	}

	handlePick(){
		let random = Math.floor(Math.random() * this.props.options.length);
		let value = this.props.options[random];
		alert(value);
	}

	render(){
			console.log(this.props.options.length);
		return (
			<div>
				<button onClick={this.handlePick} disabled={!this.props.options.length > 0}>What should we do?</button>
			</div>
		);
	}
}