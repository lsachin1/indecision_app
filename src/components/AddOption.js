import React from 'react';

export const AddOption = (props) => {
	return (
			<div>
				{props.error && <p>{props.error}</p>}
				<form onSubmit={props.formSubmit}>
					<input type="text" name="option" />
					<button>Add Option</button>
				</form>		
			</div>
		);
}