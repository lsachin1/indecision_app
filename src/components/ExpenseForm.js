import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import { addExpense } from '../actions/expenses';

const now = moment();

export default class ExpenseForm extends React.Component {

	constructor(props){

		super(props);

		this.state = {
			description: props.expense ? props.expense.description : '',
			notes: props.expense ? props.expense.notes : '',
			amount: props.expense ? (props.expense.amount / 100).toString() : '',
			createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
			focused: false,
			error: ''
		};

	}
	
	descriptionChange = (e) => {
		const description = e.target.value;
		this.setState({
			description
		})
	};

	onNoteChange = (e) => {
		const notes = e.target.value;
		//e.persist();
		this.setState({
			notes
		})
	};

	onAmountChange = (e) => {
		const amount = e.target.value;

		if(!amount || amount.toString().match(/^\d{1,}(\.\d{0,2})?$/)){
			this.setState({
				amount
			})
		}
	};

	onDateChange = (createdAt) => {
		if(createdAt){
			this.setState({
				createdAt
			})
		}
		
	}

	onFocusChange = ({ focused}) => {
		this.setState({
			focused
		})
	}

	onSubmit = (e) => {
		e.preventDefault();
		this.state.amount = this.state.amount || 0;
		if(!this.state.description){
			this.setState({
				error: 'Please enter description'
			})
		}else{
			this.setState({
				error: ''
			})

			this.props.onSubmit({
				description: this.state.description,
				amount: parseFloat(this.state.amount, 10) * 100,
				createdAt: this.state.createdAt.valueOf(),
				notes: this.state.notes
			});
			//console.log('xxxx');
		}
	}

	render() {
		return (
			<div>
			    {this.state.error && <p>{this.state.error}</p>}
				<form onSubmit={this.onSubmit}>
					<input 
						type="text"
						placeholder="description"
						autoFocus
						value={this.state.description}
						onChange={this.descriptionChange}
					/ >
					<input 
						type="number"
						placeholder="Amount"
						value={this.state.amount}
						onChange={this.onAmountChange}
					 />
					 <SingleDatePicker 
					 	date={this.state.createdAt}
					 	onDateChange={this.onDateChange}
					 	focused={this.state.focused}
					 	onFocusChange={this.onFocusChange}
					 	numberOfMonths={1}
					 	isOutsideRange={() => false}
					 />
					 <textarea
					 	placeholder="Add a note for the expense"
					 	value={this.state.notes}
					 	onChange={this.onNoteChange}
					 >
					 </textarea>
					 <button>{this.state.description ? 'Update Expense' : 'Add Expense'}</button>
					 
				</form>
			</div>
		)
	}
}