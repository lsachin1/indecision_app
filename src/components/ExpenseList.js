import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import { getVisibleExpenses } from '../selectors/expenses';

export const ExpenseList = (props) => {
	return(
		<div>
			<h1>Expense List</h1>
			{
				props.expenses.length === 0 ? ( <p>No Expenses</p> ) :
				(
					props.expenses.map((exp)=>{
						return <ExpenseListItem key={exp.id} {...exp} />
					})
				) 
			}
		</div>
	)
}

const mappedStateToProps = (state) => {
	return {
		expenses: getVisibleExpenses(state.expenses, state.filters)
	}
};

export default connect(mappedStateToProps)(ExpenseList);