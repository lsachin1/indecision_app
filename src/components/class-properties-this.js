class OldSyntax {
	constructor(){
		this.name="Sachin";
		this.getName = this.getName.bind(this);
	}

	getName(){
		console.log(this.name);
	}
}

const oldSyntax = new OldSyntax();
console.log(oldSyntax);
const getName = oldSyntax.getName;

console.log(getName());

class NewSyntax {
	name = 'Jhon';
	getName = () => {
		console.log(this.name);
	}


}

const newSyntax = new NewSyntax();
console.log(newSyntax);
const getnew = newSyntax.getName;
console.log(getnew());
