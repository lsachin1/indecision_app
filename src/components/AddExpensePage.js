import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export const AddExpensePage = (props) => {
	return (
		<div>
			<ExpenseForm onSubmit={(expense)=>{
				props.onSubmit(expense);
				props.history.push("/");
			}}/>
		</div>
	)
}

const mapDispatchToProps = (dispatch) => {
	return {
		onSubmit: (expense) =>  dispatch(startAddExpense(expense))
	}
}

export default connect(null, mapDispatchToProps)(AddExpensePage);




