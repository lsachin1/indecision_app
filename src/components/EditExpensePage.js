import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense }  from '../actions/expenses';


export const EditExpensePage = (props) => {
	//console.log(props);
	return (
		<div>
			<ExpenseForm expense={props.expense} onSubmit={(expense)=>{
				console.log("updated", expense);
				props.onSubmit(props.expense.id, expense);
				props.history.push("/");
			}}/>
		</div>
	)
}

const mappedToState = (state, props) => {
	return {
		expense: state.expenses.find((expense)=>{
			return expense.id === props.match.params.id
		})
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		onSubmit: (id, expense) => dispatch(editExpense(id, expense))
	}
}
export default connect(mappedToState, mapDispatchToProps)(EditExpensePage);