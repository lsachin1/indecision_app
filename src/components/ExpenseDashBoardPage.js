import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import Timer from './Timer';

export const ExpenseDashBoardPage = () => (
		<div>
			<Timer />
			<ExpenseListFilters />
			<ExpenseList />
		</div>
);