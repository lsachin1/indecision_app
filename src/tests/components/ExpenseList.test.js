import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';

const expense = [
	{
		id: 1,
		description: 'Gum',
		notes: '',
		amount: 195,
		createdAt: 0
	},{
		id: 2,
		description: 'Rent',
		notes: '',
		amount: 109500,
		createdAt: -1000
	},{
		id: 3,
		description: 'credit card',
		notes: '',
		amount: 4500,
		createdAt: 1000
	}
];

test('should render the expenseList from expense', ()=>{
	const wrapper = shallow(<ExpenseList expenses={expense} />);
	expect(wrapper).toMatchSnapshot();
})

test('should render the expenseList with no expense message', ()=>{
	const wrapper = shallow(<ExpenseList expenses={[]} />);
	expect(wrapper).toMatchSnapshot();
})