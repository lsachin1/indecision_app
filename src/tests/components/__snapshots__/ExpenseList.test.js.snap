// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expenseList from expense 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <Connect(ExpenseListItem)
    amount={195}
    createdAt={0}
    description="Gum"
    id={1}
    key="1"
    notes=""
  />
  <Connect(ExpenseListItem)
    amount={109500}
    createdAt={-1000}
    description="Rent"
    id={2}
    key="2"
    notes=""
  />
  <Connect(ExpenseListItem)
    amount={4500}
    createdAt={1000}
    description="credit card"
    id={3}
    key="3"
    notes=""
  />
</div>
`;

exports[`should render the expenseList with no expense message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No Expenses
  </p>
</div>
`;
